admin/helpers/check-window.js

```diff
@@ -0,0 +1,110 @@
+    liveCheckStartDateDisabled: currentDate.isSameOrAfter(checkWindowData.checkStartDate),
+    liveCheckEndDate: checkWindowData.checkEndDate,
+    liveCheckEndDay: checkWindowData.checkEndDate.format('D'),
+    liveCheckEndMonth: checkWindowData.checkEndDate.format('MM'),
+    liveCheckEndYear: checkWindowData.checkEndDate.format('YYYY'),
+    liveCheckEndDateDisabled: currentDate.isSameOrAfter(checkWindowData.checkEndDate),
+  }
+  checkWindowViewData.adminPeriodDisabled = checkWindowViewData.adminStartDateDisabled && checkWindowViewData.adminEndDateDisabled
+  checkWindowViewData.familiarisationCheckPeriodDisabled = checkWindowViewData.familiarisationCheckStartDateDisabled && checkWindowViewData.familiarisationCheckEndDateDisabled
+  checkWindowViewData.liveCheckPeriodDisabled = checkWindowViewData.liveCheckStartDateDisabled && checkWindowViewData.liveCheckEndDateDisabled
+  checkWindowViewData.pastCheckWindow = checkWindowViewData.adminPeriodDisabled && checkWindowViewData.familiarisationCheckPeriodDisabled && checkWindowViewData.liveCheckPeriodDisabled
+  return checkWindowViewData
+}
+
+/**
+ * Prepare validated data for submission
+ * @param {Object} requestData
+ * @param {Number} checkWindowId
+ * @returns {Object} Formatted check window data
+ */
+checkWindowHelper.prepareSubmissionData = (requestData, checkWindowId = null) => {
+  const checkWindowData = {}
+  if (checkWindowId) {
+    checkWindowData.id = checkWindowId
+  }
+  checkWindowData.name = requestData.checkWindowName
+  checkWindowData.adminStartDate =
+    dateService.createUTCFromDayMonthYear(
+      requestData['adminStartDay'],
+      requestData['adminStartMonth'],
+      requestData['adminStartYear']
+    )
+  checkWindowData.adminEndDate =
+    dateService.createUTCFromDayMonthYear(
+      requestData['adminEndDay'],
+      requestData['adminEndMonth'],
+      requestData['adminEndYear']
+    )
+  checkWindowData.familiarisationCheckStartDate =
+    dateService.createUTCFromDayMonthYear(
+      requestData['familiarisationCheckStartDay'],
+      requestData['familiarisationCheckStartMonth'],
+      requestData['familiarisationCheckStartYear']
+    )
+  checkWindowData.familiarisationCheckEndDate =
+    dateService.createUTCFromDayMonthYear(
+      requestData['familiarisationCheckEndDay'],
+      requestData['familiarisationCheckEndMonth'],
+      requestData['familiarisationCheckEndYear']
+    )
+  checkWindowData.checkStartDate =
+    dateService.createUTCFromDayMonthYear(
+      requestData['liveCheckStartDay'],
+      requestData['liveCheckStartMonth'],
+      requestData['liveCheckStartYear']
+    )
+  checkWindowData.checkEndDate =
+    dateService.createUTCFromDayMonthYear(
+      requestData['liveCheckEndDay'],
+      requestData['liveCheckEndMonth'],
+      requestData['liveCheckEndYear']
+    )
+  checkWindowData.checkEndDate.set({ hour: 22, minute: 59, second: 59 })
+  checkWindowData.familiarisationCheckEndDate.set({ hour: 22, minute: 59, second: 59 })
```

Comment:
whats the significance of 1 second before 11pm?  maybe put in a variable with a descriptive name.  i'm curious as to what this is for!

Reply:
Originally this was set at 23:59:59 as the last second of the day to cover the full duration of last check window date.
I noticed with BST time change that db was storing it as 00:59:59 of the next day.
I added this to prevent this overlap.

Reply:
@GeorgeChatzigiannis doesn't that mean its effectively wrong for roughly half the year though?
we baseline in UTC, so can accommodate summertime differences

Reply:
@GuyHarwood I noticed that there was no issue with the last second of the day when storing on db level so I made that change.
To answer your question the previous implementation was also moving the time to the last hour of the day so that the last day would be included in the check window period.

Reply:
@GuyHarwood I noticed that there was no issue with the last second of the day when storing on db level so I made that change.
To answer your question the previous implementation was also moving the time to the last hour of the day so that the last day would be included in the check window period.